### ISSUE TITLE
Support more type and relationship

### ISSUE THREAD

**Issue by ele828** [ele828/tamarillo#4]
Typescript can not support complex type reflection, I consider supporting it by this way:

``` typescript
@Entity()
class Person extends Model {

  @Column()
  @OneToMany(type => Person, person => person.id)
  friends: Person[];

}

```

**Comment by ele828** [comment]
Cauz for now, Typescript doesn't support complex type reflection. The only way to handle this issue is to support Types and Relationships in Decorators, it looks like:

``` typescript
@Entity(alias?)
class Foo {
    @Column(alias? | Type?)
    bucket: Type;
}

```

This is really important, only when we finish this, the Torm.js can be used. This is the core features of Torm.js.

**Comment by yfwz100** [comment]
Firstly,

``` typescript
@OneToMany(type => Person, person => person.id)
```

Don't know what you mean in the example. `person.id` seems to refer to a undefined object `person`. It's not a correct syntax.

Why not try

``` typescript
@OneToMany(type => Person, person => "id")
```

And you mention `@Column(alias? | Type?)`, what's the meaning of alias and type? I see that type definition is given by variable definition.

**Comment by ele828** [comment]
@yfwz100 yeah. you're right! What I want to do here is to set a foreign key.

`@Column(alias? | Type?)` This means the first argument can be Alias or Complex Type. Some types like float double (we don't have this in JS) or customized Type cannot be reflected for now in Typescript. Do you have any better ideas for this?

**Comment by yfwz100** [comment]
Got the idea. In Mongoose-like ORM in javascript, the complex type is modeled as a new class. I think we can adopt this. The drafted syntax could be like

```
@Column(type=Float, alias="col_name")
```

where `Float` is a custom class defined by us (so it should be imported before used). The idea of using custom type instead of raw string is to facilitate the development with syntax check at compile time.

**Comment by ele828** [comment]
@yfwz100 , that's exactly what I want
